src/cmds.c

```diff
@@ -2236,7 +2236,7 @@ static void cmd_su(struct userrec *u, int idx, char *par)
         dcc[idx].u.chat->su_nick = get_data_ptr(strlen(dcc[idx].nick) + 1);
         strcpy(dcc[idx].u.chat->su_nick, dcc[idx].nick);
         dcc[idx].user = u;
-        strcpy(dcc[idx].nick, par);
+        strncpyz(dcc[idx].nick, par, NICKLEN);
```

Comment:
Going off memory, some of these are not declared in this space and thus sizeof doesn't work- I'm assuming that is the case for these